File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/provider/aop/SendMqMessage.java
package org.xyattic.eventual.consistency.support.core.provider.aop;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author wangxing
 * @create 2020/4/1
 */
@Target({ElementType.METHOD, ElementType.ANNOTATION_TYPE, ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Inherited
public @interface SendMqMessage {

    String persistenceName() default "";

    String transactionManager() default "";

}

File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/provider/aop/SendMqMessage.kt
package org.xyattic.eventual.consistency.support.core.provider.aop

import java.lang.annotation.Inherited

/**
 * @author wangxing
 * @create 2020/4/1
 */
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.ANNOTATION_CLASS,
    AnnotationTarget.ANNOTATION_CLASS,
    AnnotationTarget.CLASS
)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Inherited
annotation class SendMqMessage(
    val persistenceName: String = "",
    val transactionManager: String = "",
    val delayedSend: Boolean = false
)